dbt_subprojects/solana/models/_sector/dex/stabble/stabble_solana_base_trades.sql

```diff
@@ -0,0 +1,191 @@
+        account_user_token_in,      
+        account_user_token_out,    
+        account_vault_token_in AS token_sold_vault,
+        account_vault_token_out AS token_bought_vault
+    FROM {{ source('stable_swap_solana', 'stable_swap_call_swap_v2') }}
+    WHERE call_block_time >= CURRENT_TIMESTAMP - INTERVAL '7' DAY
+    {% if is_incremental() %}
+    AND {{incremental_predicate('call_block_time')}}
+    {% endif %}
+
+    UNION ALL
+
+    -- Weighted Swap V1
+    SELECT
+        'weighted_swap_v1' AS swap_type,
+        call_block_time AS block_time,
+        call_block_slot AS block_slot,
+        'stabble' AS project,
+        1 AS version,
+        'solana' AS blockchain,
+        CASE WHEN call_is_inner = FALSE THEN 'direct' ELSE call_outer_executing_account END AS trade_source,
+        amount_in AS token_sold_amount_raw,
+        account_pool AS pool_id,
+        call_tx_signer AS trader_id,
+        call_tx_id AS tx_id,
+        call_outer_instruction_index AS outer_instruction_index,
+        COALESCE(call_inner_instruction_index, 0) AS inner_instruction_index,
+        call_tx_index AS tx_index,
+        CAST(NULL AS VARCHAR) AS token_sold_mint_address,      
+        CAST(NULL AS VARCHAR) AS token_bought_mint_address,    
+        account_user_token_in,      
+        account_user_token_out,    
+        account_vault_token_in AS token_sold_vault,
+        account_vault_token_out AS token_bought_vault
+    FROM {{ source('stable_swap_solana', 'weighted_swap_call_swap') }}
+    WHERE call_block_time >= CURRENT_TIMESTAMP - INTERVAL '7' DAY
+    {% if is_incremental() %}
+    AND {{incremental_predicate('call_block_time')}}
+    {% endif %}
+
+    UNION ALL
+
+    -- Weighted Swap V2
+    SELECT
+        'weighted_swap_v2' AS swap_type,
+        call_block_time AS block_time,
+        call_block_slot AS block_slot,
+        'stabble' AS project,
+        1 AS version,
+        'solana' AS blockchain,
+        CASE WHEN call_is_inner = FALSE THEN 'direct' ELSE call_outer_executing_account END AS trade_source,
+        amount_in AS token_sold_amount_raw,
+        account_pool AS pool_id,
+        call_tx_signer AS trader_id,
+        call_tx_id AS tx_id,
+        call_outer_instruction_index AS outer_instruction_index,
+        COALESCE(call_inner_instruction_index, 0) AS inner_instruction_index,
+        call_tx_index AS tx_index,
+        account_mint_in AS token_sold_mint_address,      
+        account_mint_out AS token_bought_mint_address,    
+        account_user_token_in,      
+        account_user_token_out,    
+        account_vault_token_in AS token_sold_vault,
+        account_vault_token_out AS token_bought_vault
+    FROM {{ source('stable_swap_solana', 'weighted_swap_call_swap_v2') }}
+    WHERE call_block_time >= CURRENT_TIMESTAMP - INTERVAL '7' DAY
+    {% if is_incremental() %}
+    AND {{incremental_predicate('call_block_time')}}
+    {% endif %}
+)
+
+, transfers AS (
+    SELECT DISTINCT
```

Comment:
what happens if we don't distinct here, the join condition seems so specific that I am suprised you got multiple hits anywhere

Reply:
essentially what's the explanation for having to distinct, is there cases where the vault transfers tokens to the user twice?